packages/alfa-html/src/alphabet.ts

```diff
@@ -126,6 +130,27 @@ const data: Pattern = (stream, emit, state) => {
 };
 
 /**
+ * 8.2.4.4
+ * @see https://www.w3.org/TR/html/syntax.html#script-data-state
+ */
+const scriptData: Pattern = (stream, emit, state) => {
+  const char = stream.peek(0);
+  stream.advance(1);
+  switch (char) {
+    case Char.LessThanSign:
+      return scriptDataLessThanSign;
+    case Char.Null:
+      emit({ type: TokenType.Character, data: fromCharCode(65533) });
+      break;
+    case null:
+      return Command.End;
+    default:
+      emit({ type: TokenType.Character, data: fromCharCode(char) });
+  }
+};
```

Comment:
Make sure to leave a little room between expressions and statements; the above looks very cramped, making it harder to get an overview of what's going on. I'd do:

```ts
const scriptData: Pattern = (stream, emit, state) => {
  const char = stream.peek(0);
  stream.advance(1);

  switch (char) {
    case Char.LessThanSign:
      return scriptDataLessThanSign;

    case Char.Null:
      emit({ type: TokenType.Character, data: fromCharCode(65533) });
      break;

    case null:
      return Command.End;

    default:
      emit({ type: TokenType.Character, data: fromCharCode(char) });
  }
};
```

Whitespace is your friend!

Reply:
This applies to everything below as well.